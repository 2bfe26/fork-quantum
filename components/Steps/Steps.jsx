import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  display: table;
  width: 100%;
`;

const ProgressBar = styled.ul`
  padding: 0;
  display: flex;
  justify-content: space-between;
`;

const Steps = ({ children }) => (
  <Wrapper>
    <ProgressBar>
      { children }
    </ProgressBar>
  </Wrapper>
);

Steps.propTypes = {
  /** Step Component list */
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
  ]).isRequired,
};

export default Steps;
