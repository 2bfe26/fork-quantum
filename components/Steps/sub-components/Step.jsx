import React from 'react';
import PropTypes from 'prop-types';

const Step = ({ title, active }) => {
  const isActive = active ? 'active' : '';

  return (
    <React.Fragment>
      <li className={isActive}>
        <p>{title}</p>
      </li>
    </React.Fragment>
  );
};

Step.propTypes = {
  title: PropTypes.string.isRequired,
  // icon: PropTypes.string,
  active: PropTypes.bool,
};

Step.defaultProps = {
  active: false,
};

export default Step;
