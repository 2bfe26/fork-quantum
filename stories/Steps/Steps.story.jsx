import React from 'react';
import { linkTo } from '@storybook/addon-links';
import { storiesOf } from '@storybook/react';
import Heading from '../../.storybook/components/Heading';
import AutoPropsApi from '../../.storybook/components/AutoPropsApi';
import HowToImport from '../../.storybook/components/HowToImport';
import Example from '../../.storybook/components/Example';
import { TabbedView, Tab } from '../../.storybook/components/TabbedView';
import { Col, Row } from '../../components/Grid';
import Button from '../../components/Button';
import CodeExample from '../../.storybook/components/CodeExample';
import Atom from '../static/atom.svg';
import { Steps, Step } from '../../components/Steps';

const StepExample = (
  <Steps>
    <Step title="Criação de conta" icon="person" status="active" />
    <Step title="Plano e forma de pagamento" icon="credit_card" />
    <Step title="Preencher o currículo" icon="description" />
  </Steps>
);

const stories = storiesOf('10. Steps', module);

stories
  .add('Steps', () => (
    <Heading name="Steps" image={Atom}>
      <TabbedView>
        <Tab title="Usage">
          <HowToImport importModules="Steps, Step" />
          <p>We provide two components to use steps: <code>{ '<Steps>' }</code> and <code>{ '<Step title="Example">' }</code> must be used together.</p>
          <p>The <code>{ '<Step>' }</code> must have a required title prop and his current status prop, that can be one of:</p>

          <ul>
            <li><code>&nbsp;</code> <small>(empty string)</small></li>
            <li>active</li>
            <li>done</li>
          </ul>

          <p>You can pass an icon to be displayed with the step component, see the example below:</p>
          <small>You can see the list of enable icons <Button onClick={linkTo('1. Foundation', 'Icons')}>here</Button></small>

          <h3>A simple 3 steps component</h3>
          <Row>
            <Col desktop={4} phone={12}>
              <CodeExample component={StepExample} />
            </Col>
            <Col desktop={8} phone={12}>
              {StepExample}
            </Col>
          </Row>
        </Tab>
        <Tab title="API">
          <AutoPropsApi component={Step} title="Step" />
          <AutoPropsApi component={Steps} title="Steps" />
        </Tab>
        <Tab title="Examples">
          <h3>First step active</h3>
          <Example component={
            <Steps>
              <Step title="Criação de conta" icon="person" status="active" />
              <Step title="Plano e forma de pagamento" icon="credit_card" />
              <Step title="Preencher o currículo" icon="description" />
            </Steps>
            }
          />
          <h3>First step done, second step active</h3>
          <Example component={
            <Steps>
              <Step title="Criação de conta" icon="person" status="done" />
              <Step title="Plano e forma de pagamento" status="active" icon="credit_card" />
              <Step title="Preencher o currículo" icon="description" />
            </Steps>
            }
          />
          <h3>First and second steps done, third step active</h3>
          <Example component={
            <Steps>
              <Step title="Criação de conta" icon="person" status="done" />
              <Step title="Plano e forma de pagamento" status="done" icon="credit_card" />
              <Step title="Preencher o currículo" status="active" icon="description" />
            </Steps>
            }
          />
          <h3>All steps done</h3>
          <Example component={
            <Steps>
              <Step title="Criação de conta" icon="person" status="done" />
              <Step title="Plano e forma de pagamento" status="done" icon="credit_card" />
              <Step title="Preencher o currículo" status="done" icon="description" />
            </Steps>
            }
          />
        </Tab>
      </TabbedView>
    </Heading>
  ));
