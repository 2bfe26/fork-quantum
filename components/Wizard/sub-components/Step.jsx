import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Icon from '../../Icon';
import Colors from '../../Colors';
import { BREAKPOINTS } from '../../Grid/sub-components';

const statusFontColor = ({ status }) => {
  const statusColors = {
    normal: Colors.GREY['100'],
    active: Colors.WHITE,
    done: Colors.PRIMARY['500'],
  };

  return `color: ${statusColors[status] || statusColors.normal};`;
};

const statusStepColor = ({ status }) => {
  const stepColors = {
    normal: `
      color: ${Colors.GREY['400']};

      &:after {
        background-color: ${Colors.GREY['100']};
      }

      &:before {
        background-color: ${Colors.WHITE};
        border: 2px solid ${Colors.GREY['100']};
      }
    `,
    active: `
      color: ${Colors.PRIMARY['500']};

      &:after {
        background-color: ${Colors.PRIMARY['500']};
      }

      &:before {
        background-color: ${Colors.PRIMARY['500']};
        border: 2px solid ${Colors.PRIMARY['500']};
      }
    `,
    done: `
      color: ${Colors.GREY['400']};

      &:after {
        background-color: ${Colors.PRIMARY['500']};
      }

      &:before {
        background-color: ${Colors.WHITE};
        border: 2px solid ${Colors.PRIMARY['500']};
      }
    `,
  };

  return stepColors[status] || stepColors.normal;
};

const StepIcon = styled(Icon)`
  position: relative;
  top: -42px;

  ${statusFontColor}
`;

const ProgressStep = styled.li`
  list-style-type: none;
  flex-grow: 1;
  flex: 1;
  font-size: 12px;
  position: relative;
  text-align: center;
  margin-top: 50px;

  @media (max-width: ${BREAKPOINTS.phone}px) {
    font-size: 10px;
  }

  &:after {
    position: absolute;
    top: 15px;
    left: -50%;
    width: 100%;
    height: 2px;
    content: '';
    z-index: -1;
  }

  &:before {
    width: 36px;
    height: 36px;
    line-height: 30px;
    content: '';
    display: block;
    text-align: center;
    margin: 0 auto 10px;
    border-radius: 50%;
  }

  &:first-child:after {
    content: none;
  }

  ${statusStepColor}
`;

const Title = styled.p`
  position: absolute;
  top: -24px;
  margin: 0 auto;
  padding: 4px;
  overflow: hidden;
  width: 100%;
  word-wrap: break-word;

  @media (min-width: ${BREAKPOINTS.phone + 1}px) and (max-width: ${BREAKPOINTS.tablet}px) {
    top: -34px;
  }
  @media (max-width: ${BREAKPOINTS.phone}px) {
    top: -30px;
  }
`;

const statusTypes = ['', 'active', 'done'];

const Step = ({ title, status, icon }) => (
  <ProgressStep status={status}>
    <Title>{ title }</Title>
    { icon && <StepIcon status={status} name={icon} /> }
  </ProgressStep>
);

Step.propTypes = {
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
  icon: PropTypes.string,
  status: PropTypes.oneOf(statusTypes),
};

Step.defaultProps = {
  status: '',
  icon: '',
};

StepIcon.propTypes = {
  status: PropTypes.oneOf(statusTypes),
};

export default Step;
