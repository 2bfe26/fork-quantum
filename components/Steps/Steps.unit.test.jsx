import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Steps from './Steps';
import Step from '../../components/Steps/sub-components/Step';

describe('Steps Component', () => {
  const component = (
    <Steps>
      <Step title="Criação de conta" icon="person" status="done" />
      <Step title="Plano Forma de pagamento" icon="credit_card" status="active" />
    </Steps>
  );
  const wrapper = shallow(component);

  it('Should match the snapshot', () => {
    const tree = renderer.create((component)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should verify if Steps contains children', () => {
    expect(wrapper.find('Step')).toHaveLength(2);
  });
});
