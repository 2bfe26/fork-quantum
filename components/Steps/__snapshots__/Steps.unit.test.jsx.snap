// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps Component Should match the snapshot 1`] = `
<div
  className="sc-bdVaJa cOVzqF"
>
  <ul
    className="sc-bwzfXH ceHnQp"
  >
    <li
      className="sc-gzVnrw ideYlH"
    >
      <p>
        Criação de conta
      </p>
      <span
        aria-hidden="true"
        className="material-icons MuiIcon-root-1 sc-bZQynM eJOqQW"
        status="done"
        style={
          Object {
            "color": "",
          }
        }
      >
        person
      </span>
    </li>
    <li
      className="sc-gzVnrw GPMNN"
    >
      <p>
        Plano Forma de pagamento
      </p>
      <span
        aria-hidden="true"
        className="material-icons MuiIcon-root-1 sc-bZQynM crTbuC"
        status="active"
        style={
          Object {
            "color": "",
          }
        }
      >
        credit_card
      </span>
    </li>
  </ul>
</div>
`;
